core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -1084,8 +1084,8 @@ void populate2() {
       final RexCallImplementor value = new ValueConstructorImplementor();
       define(MAP_VALUE_CONSTRUCTOR, value);
       define(ARRAY_VALUE_CONSTRUCTOR, value);
-      defineMethod(ARRAY, BuiltInMethod.ARRAYS_AS_LIST.method, NullPolicy.NONE);
-      defineMethod(MAP, BuiltInMethod.MAP.method, NullPolicy.NONE);
+      defineMethod(ARRAY, BuiltInMethod.ARRAYS_AS_LIST.method, NullPolicy.STRICT);
```

Comment:
these two are definitely wrong

Reply:
Again, I think this is the same issue. Unless I have misunderstood/misread the linq code that consumers this the guards here are for the array references, and not their contents.